### ISSUE TITLE
Colorization of mono games

### ISSUE THREAD

**Issue by eadmaster** [libretro/beetle-wswan-libretro#79]
I am trying to hack a colorization option for mono games, similar to [what i've done with the VirtualBoy some years ago](https://eadmaster.altervista.org/pub/index.php?page=shaders&theme=tango).

This looks more tricky than my previous attempt since this system seems to be using a dynamic grayscale palette.

Btw i'd like to hack directly the color palette into the code, so i can use it in the 3DS which has no shaders support.

While i am still studying the code, do you think this is doable?

**Comment by eadmaster** [comment]
little update: starting to get some colors already!

![Kaze no Klonoa - Moonlight Museum (J)  M t1   T+Eng0 511_PowerStone05 -211126-083312](https://user-images.githubusercontent.com/925171/143543990-d1deeb11-4de1-4fab-a92a-dc7b64e5f3cb.png)

![Kaze no Klonoa - Moonlight Museum (J)  M t1   T+Eng0 511_PowerStone05 -211126-083237](https://user-images.githubusercontent.com/925171/143544001-30abd0d8-278e-4def-8c2f-37bd196977c7.png)


I guess the next step would be having diff. palettes for the sprites vs the bgs...

**Comment by twinaphex** [comment]
looks cool, very promising.

**Comment by hizzlekizzle** [comment]
Nice! looks like it is indeed doable :)

Are you doing a straight interpolation between two colors?

**Comment by eadmaster** [comment]
> Are you doing a straight interpolation between two colors?

No, actually i am just [boosting 2 color channels in this palette init function](https://github.com/eadmaster/beetle-wswan-libretro/blob/e5a76b5491c89ba811391c339762fe7d00a221ad/mednafen/wswan/gfx.c#L291).

When combined with the current "WS Colorization " option, you get even more variations than the 2 i've posted, although not all of them looks good with this game.

As i've said, i think i can get even better results if i manage to differentiate the palettes used for the sprites vs the BGs.
